pkg/envinfo/storage.go

```diff
@@ -56,39 +56,63 @@ func (ei *EnvInfo) GetStorage(name string) (*localConfigProvider.LocalStorage, e
 
 // CreateStorage sets the storage related information in the local configuration
 func (ei *EnvInfo) CreateStorage(storage localConfigProvider.LocalStorage) error {
+	//initialize volume mount and volume container
+	vm := []devfilev1.VolumeMount{
+		{
+			Name: storage.Name,
+			Path: storage.Path,
+		},
+	}
+	vc := []devfilev1.Component{{
+		Name: storage.Name,
+		ComponentUnion: devfilev1.ComponentUnion{
+			Volume: &devfilev1.VolumeComponent{
+				Volume: devfilev1.Volume{
+					Size: storage.Size,
+				},
+			},
+		},
+	}}
+	volumeExists := false
 	// Get all the containers in the devfile
 	containers, err := ei.GetContainers()
 	if err != nil {
 		return err
 	}
 
-	// Add volumeMount to all containers in the devfile
+	// Add volumeMount to all containers if no container is specified else to specified container(s) in the devfile
 	for _, c := range containers {
-		if err := ei.devfileObj.Data.AddVolumeMounts(c.Name, []devfilev1.VolumeMount{
-			{
-				Name: storage.Name,
-				Path: storage.Path,
-			},
-		}); err != nil {
-			return err
+		if storage.Container == "" || (storage.Container != "" && c.Name == storage.Container) {
+			if err := ei.devfileObj.Data.AddVolumeMounts(c.Name, vm); err != nil {
+				return err
+			}
```

Comment:
We need to also check if the given container exists or not. 

```
[mrinaldas@localhost project]$ odo storage create --container fredajdkjsadkj
 ✓  Added storage nodejs-project-yvzb-ffnp to nodejs-project-yvzb

Please use `odo push` command to make the storage accessible to the component
```
In the above example, `fredajdkjsadkj` is an invalid container.

Reply:
The check was added and will be push shortly